spi/transfer-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/transfer/TransferProcess.java

```diff
@@ -197,60 +209,52 @@ public void transitionInProgressOrStreaming() {
     public void transitionInProgress() {
         if (type == Type.CONSUMER) {
             // the consumer must first transition to the request/ack states before in progress
-            transition(TransferProcessStates.IN_PROGRESS, TransferProcessStates.REQUESTED, TransferProcessStates.REQUESTED_ACK);
+            transition(IN_PROGRESS, REQUESTED);
         } else {
             // the provider transitions from provisioned to in progress directly
-            transition(TransferProcessStates.IN_PROGRESS, TransferProcessStates.REQUESTED, TransferProcessStates.PROVISIONED);
+            transition(IN_PROGRESS, REQUESTED, PROVISIONED);
         }
     }
 
     public void transitionStreaming() {
         if (type == Type.CONSUMER) {
             // the consumer must first transition to the request/ack states before in progress
-            transition(TransferProcessStates.STREAMING, TransferProcessStates.REQUESTED_ACK);
+            transition(STREAMING, REQUESTED);
```

Comment:
Since we're using pessimistic locking with expiring leases and assuming idempotency, I think a transition into the same state should always be allowed, except maybe for terminal states like CANCELLED, ERROR,  COMPLETED. This is probably the same for all `transition`-methods